src/Domain/Mesh.hpp

```diff
@@ -123,6 +123,14 @@ class Mesh {
    */
   size_t number_of_grid_points() const noexcept { return extents_.product(); }
 
+  /*!
+   * \brief Returns the 1-dimensional index corresponding to the `Dim`
+   * dimensional `index`.
+   */
+  size_t storage_index(const std::array<size_t, Dim>& index) const noexcept {
```

Comment:
Add an overload that takes an `Index<Dim>`? Why even have the `std::array` overload (same in `Index.hpp`)?

Reply:
Partly because I'm still confused why `Index` doesn't inherit off of `std::array` and I wanted `storage_index({{a,b,c}})` to work, though that may work with `Index` too, I'll test.